runtime/runtime/src/lib.rs

```diff
@@ -128,7 +128,7 @@ pub struct ActionResult {
     pub result: Result<ReturnData, ActionError>,
     pub logs: Vec<LogEntry>,
     pub new_receipts: Vec<Receipt>,
-    pub validator_proposals: Vec<ValidatorStake>,
+    pub validator_proposals: BTreeMap<AccountId, ValidatorStake>,
```

Comment:
In theory we don't need sorted map, but something like a linked_hash_map, which Rust doesn't really have :)

Reply:
yep :)

Reply:
The issue with `BTreeMap` is we're changing the spec by reordering proposals based on lexicographical order of AccountID, which is not great. Ideally, we shouldn't change it. Better to keep it as Vec, but dedup with hashmap at the end.

Reply:
> by reordering proposals based on lexicographical order of AccountID, which is not great

Why does it matter?

Reply:
Because it was not required before, but now it'll have to be enforced.

Reply:
Okay